### PR TITLE
ApiNodeContentClient 

### PR DESCRIPTION
**Pull request by kerkhofsd**

### CODE REVIEW

**Review by JanVdloock — APPROVED** [review]

File: alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/ApiNodeContentSpringClient.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public ApiNodeContentSpringClient(AlfrescoProperties alfrescoProperties, RestTemplate restTemplate) {
+        this.url = UriComponentsBuilder.fromHttpUrl(alfrescoProperties.getUrl())
+                .path("/service/api/node/content")
+                .toUriString();
+
+        this.restClient = restTemplate;
+    }
+
+    @Override
+    public boolean hasContent(String nodeRef) {
```
Comment (tgeens) [inline alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/ApiNodeContentSpringClient.java:39]: don't you need to check which content-property ?

File: alfresco-webscripts-client/alfresco-webscripts-ditto/src/main/java/eu/xenit/alfresco/webscripts/client/ditto/ApiNodeContentFakeClient.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        if (!this.nodeView.getNode(nodeRef).isPresent() ||
+                !this.nodeView.getNode(nodeRef).get().getProperties().getContentData().isPresent()) {
+            throw new RuntimeException("Node not available or has no content: " + nodeRef);
+        }
+
+        try (InputStream inputStream = contentView.getContent(nodeRef)
+                .orElseThrow(() -> new IllegalStateException("Node " + nodeRef + " does not exist"))) {
+            IOUtils.copy(inputStream, outputStream);
```
Comment (todorinskiz) [inline alfresco-webscripts-client/alfresco-webscripts-ditto/src/main/java/eu/xenit/alfresco/webscripts/client/ditto/ApiNodeContentFakeClient.java:59]: Hmm in Object Storage Tools we use Spring Core, but here we don't have it... sure, Apache Commons it is then :+1:

Comment (todorinskiz) [reply alfresco-webscripts-client/alfresco-webscripts-ditto/src/main/java/eu/xenit/alfresco/webscripts/client/ditto/ApiNodeContentFakeClient.java:59]: Btw in my streams refactoring, `getContent` returns an InputStream and this copy is avoided anyways, will be needed only for Swarm SDK, unless we decide to use their REST API.

**Comment by kerkhofsd** [comment]
Updated PR with extending the Fake implementation, supporting other d:content properties as well

File: alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/ApiNodeContentSpringClient.java (hunk 1 above)
Comment (kerkhofsd) [reply alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/ApiNodeContentSpringClient.java:39]: See: `org.alfresco.repo.web.scripts.content.ContentInfo`:
```
    @Override
    public void execute(WebScriptRequest req, WebScriptResponse res) throws IOException
    {
        // create empty map of args
        Map<String, String> args = new HashMap<String, String>(0, 1.0f);
        // create map of template vars
        Map<String, String> templateVars = req.getServiceMatch().getTemplateVars();
        // create object reference from url
        ObjectReference reference = createObjectReferenceFromUrl(args, templateVars);
        NodeRef nodeRef = reference.getNodeRef();
        if (nodeRef == null)
        {
            throw new WebScriptException(HttpServletResponse.SC_NOT_FOUND, "Unable to find " + reference.toString());
        }

        // render content
        QName propertyQName = ContentModel.PROP_CONTENT;

        // Stream the content
        streamContent(req, res, nodeRef, propertyQName, false, null, null);
    }
```
that's how it works :man_shrugging:

File: alfresco-webscripts-client/alfresco-webscripts-ditto/src/main/java/eu/xenit/alfresco/webscripts/client/ditto/ApiNodeContentFakeClient.java (hunk 2 above)
Comment (kerkhofsd) [reply alfresco-webscripts-client/alfresco-webscripts-ditto/src/main/java/eu/xenit/alfresco/webscripts/client/ditto/ApiNodeContentFakeClient.java:59]: :+1:  can be refactored later.

File: alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/ApiNodeContentSpringClient.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            return false;
+        } catch (Exception e) {
+            logger.error("HEAD ../api/node/content '{}' unexpected error", nodeRef, e);
+            return false;
+        }
+
+        return true;
+    }
+
+    @SneakyThrows(UnsupportedEncodingException.class)
```
Comment (todorinskiz) [inline alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/ApiNodeContentSpringClient.java:54]: Can't we add the exception in the interface and let the consumer decide what to do with it?

Comment (kerkhofsd) [reply alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/ApiNodeContentSpringClient.java:54]: `UnsupportedEncodingException` is very implementation specific IMO

**Review by todorinskiz — COMMENTED** [review]
A few comments left behind for clarification, but overall looks good to merge!

**Review by todorinskiz — APPROVED** [review]

**Pull request merged**